### ISSUE TITLE
softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by henbucuoshanghai** [mks0601/PoseFix_RELEASE#29]
1:why  use this loss??  softmax loss with heatmaps??
2:the (x,y) of the keypoint use l1 loss,    the (x,y) from the heatmaps?\

**Comment by mks0601** [comment]
1. My groundtruth one-hot heatmap has a single blob in continuous space. For example, it has
0 0.5 0.5 1, not 0 0 1 0. Therefore, I cannot use standard cross entropy loss function, because it requires a single integer label for the groundtruth.

2. (x,y) is obtained from the heatmap using soft-argmax operation.

**Comment by Lixiuxiu2020** [comment]
Hi, @mks0601, thank you for sharing the code.
There is softmax_cross_entropy_with_logits in tensorflow, but how to implement this function in pytorch.

**Comment by mks0601** [comment]
You can simply google it.
https://stackoverflow.com/questions/46218566/pytorch-equivalence-for-softmax-cross-entropy-with-logits

**Comment by Lixiuxiu2020** [comment]
Thank you for the link.
I have a question that the input image is noly subtract mean([123.68, 116.78, 103.94]) in your code. Why not divide by the std ([0.229 * 255, 0.224 * 255, 0.225 * 255]). If I use (img - mean)/std to normalize the image in your code,will it affect the effect of network training ?

**Comment by mks0601** [comment]
I don't think it would affect the final performance significantly.

**Comment by Lixiuxiu2020** [comment]
Thank you
